iree/hal/local/loaders/system_library_loader.h

```diff
@@ -1,30 +0,0 @@
-// Copyright 2020 The IREE Authors
-//
-// Licensed under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-
-#ifndef IREE_HAL_LOCAL_LOADERS_SYSTEM_LIBRARY_LOADER_H_
-#define IREE_HAL_LOCAL_LOADERS_SYSTEM_LIBRARY_LOADER_H_
-
-#include <stdbool.h>
-#include <stdint.h>
-
-#include "iree/base/api.h"
-#include "iree/hal/local/executable_loader.h"
-
-#ifdef __cplusplus
-extern "C" {
-#endif  // __cplusplus
-
-// Creates an executable loader that can load files from platform-supported
-// dynamic libraries (such as .dylib on darwin, .so on linux, .dll on windows).
```

Comment:
Can you comment a bit on why this is going away, and how that relates to the rest of this PR? We still support loading executables from platform-supported dynamic libraries, right?

Reply:
it was never finished - the legacy loader is what's used today, and instead of keeping this in sync in its partial untested form I'm just going to cut over the legacy one into this in a single future PR